mri/reconstruct/linear.py

```diff
@@ -114,13 +116,185 @@ def l2norm(self, shape):
             the L2 norm.
         """
         # Create fake data
-        shape = numpy.asarray(shape)
+        shape = np.asarray(shape)
         shape += shape % 2
-        fake_data = numpy.zeros(shape)
+        fake_data = np.zeros(shape)
         fake_data[tuple(zip(shape // 2))] = 1
 
         # Call mr_transform
         data = self.op(fake_data)
 
         # Compute the L2 norm
-        return numpy.linalg.norm(data)
+        return np.linalg.norm(data)
+
+
+class WaveletUD2(object):
+    """The wavelet undecimated operator using pysap wrapper.
+    """
+    def __init__(self, wavelet_id=24, nb_scale=4,
+                 multichannel=False, n_cpu=1, verbose=0):
+        """Init function for Undecimated wavelet transform
+
+        Parameters
+        -----------
+        wavelet_id: int, default 24 = undecimated (bi-) orthogonal transform
+            ID of wavelet being used
+        nb_scale: int, default 4
+            the number of scales in the decomposition.
+        multichannel: bool, default False
+            Boolean value to indicate if the incoming data is from
+            multiple-channels
+        n_cpu: int, default 0
+            Number of CPUs to run on. Only applicable if multichannel=True.
+        Private Variables:
+            _has_run: Checks if the get_mr_filters was called already
+        """
+        self.wavelet_id = wavelet_id
+        self.multichannel = multichannel
+        self.nb_scale = nb_scale
+        self.n_cpu = n_cpu
+        self._opt = [
+            '-t{}'.format(self.wavelet_id),
+            '-n{}'.format(self.nb_scale),
+        ]
+        self._has_run = False
+        self.coeffs_shape = None
+
+    def _get_filters(self, shape):
+        """Function to get the Wavelet coefficients of Delta[0][0].
```

Comment:
Just as a tip for next time, you don't have to document private functions.
Now that it's here let's leave it.

Reply:
Sure